src/App.jsx

```diff
@@ -14,33 +16,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const ALPHABET_SORT_CONDITION = 'alphabet';
+  const LENGTH_SORT_CONDITION = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const [sortCondition, setSortCondition] = useState('');
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  useEffect(() => {
+    let sortedGoods = [...visibleGoods];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    switch (sortCondition) {
+      case ALPHABET_SORT_CONDITION:
+        sortedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case LENGTH_SORT_CONDITION:
+        sortedGoods.sort((a, b) => a.length - b.length);
+        break;
+      case 'reverse':
+        sortedGoods = sortedGoods.reverse();
+        break;
+      default:
+        break;
+    }
+
+    setVisibleGoods(sortedGoods);
+  }, [sortCondition, visibleGoods]);
+
+  const handleReset = () => {
+    setVisibleGoods(goodsFromServer);
+    setSortCondition('');
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortCondition !== ALPHABET_SORT_CONDITION,
+          })}
+          onClick={() => setSortCondition(ALPHABET_SORT_CONDITION)}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortCondition !== LENGTH_SORT_CONDITION,
+          })}
+          onClick={() => setSortCondition(LENGTH_SORT_CONDITION)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': sortCondition !== 'reverse',
+          })}
+          onClick={() => setSortCondition('reverse')}
+        >
+          Reverse
```

Comment:
This condition was not implemented, create a separate state for reverse with false value by default and change it onClick
![image](https://github.com/mate-academy/react_list-of-goods-js/assets/80700247/30c6c9fa-cd8b-4a2c-8b64-8b32eeffd6e7)